vrecord

```diff
@@ -1932,11 +1932,19 @@ elif [[ "${DEVICE_INPUT_CHOICE}" = 2 ]] ; then
         GRAB_INPUT+=(-i hw:"${AUDIO_DEV_NUM}")
         EXTENSION="${AUDIO_EXT}"
         FORMAT="${AUDIO_EXT}"
-    elif [[ "${OS_TYPE}" = "macOS" ]] ; then
+    elif [[ "${OS_TYPE}" = "macOS" ]] && [[ "${AUDIO_DEV_CHOICE}" != *"Blackmagic"* ]] ; then
         GRAB_INPUT+=(-f avfoundation)
         GRAB_INPUT+=(-i "none:${AUDIO_DEV_CHOICE}")
         EXTENSION="${AUDIO_EXT}"
         FORMAT="${AUDIO_EXT}"
+    elif [[ "${OS_TYPE}" = "macOS" ]] && [[ "${AUDIO_DEV_CHOICE}" == *"Blackmagic"* ]] ; then
+        GRAB_INPUT+=(-f decklink)
+        GRAB_INPUT+=(-channels 2)
+        GRAB_INPUT+=(-audio_depth 32)
+        GRAB_INPUT+=(-i "${DECKLINK_INPUT_CHOICE}")
+        GRAB_INPUT+=(-vn)
+        EXTENSION="${AUDIO_EXT}"
+        FORMAT="${AUDIO_EXT}"
```

Comment:
could use the _report_unexpected_error function is none of the conditions are tripped even though they are expected to